app/src/main/java/com/example/tasky/feature_agenda/data/repository/AgendaRepositoryImpl.kt

```diff
@@ -2,114 +2,161 @@ package com.example.tasky.feature_agenda.data.repository
 ) : AgendaRepository {
-    @RequiresApi(Build.VERSION_CODES.O)
-    override suspend fun getAgenda(fetchFromRemote: Boolean): Flow<Resource<List<AgendaItem>>> {
-        return flow {
-            emit(Resource.Loading())
 
-            val events = daos.eventDao.getEvents()
+    override suspend fun logout() {
+        api.logout()
+    }
 
-            val eventsWithAttendees = events.map {
-                daos.eventDao.getEventWithAttendees(it.eventId)
+    @RequiresApi(Build.VERSION_CODES.O)
+    override suspend fun getAgenda(): Flow<List<AgendaItem>> {
+        return combine(
+            db.eventDao.getEvents(),
+            db.reminderDao.getReminders(),
+            db.taskDao.getTasks()) { events, reminders, tasks ->
+
+            val eventWithAttendees = events.map {
+                db.eventDao.getEventWithAttendees(it.eventId)
             }
 
-            val localEvents = eventsWithAttendees.map { it.toEvent() }
-            val localTasks = daos.taskDao.getTasks().map { it.toTask() }
-            val localReminders = daos.reminderDao.getReminders().map { it.toReminder() }
+            val localEvents = eventWithAttendees.map { it.toEvent() }
+            val localReminders = reminders.map { it.toReminder() }
+            val localTasks = tasks.map { it.toTask() }
+
+            (localEvents + localReminders + localTasks).sortedBy { it.sortDate }
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    override suspend fun fetchAgendaFromRemote(): AgendaResult {
 
-            val localAgenda = localEvents + localTasks + localReminders
+        val remoteAgenda = try {
+            val response = api.getAgenda(
+                timeZone = TimeZone.getDefault().toString(),
+                time = System.currentTimeMillis()
+            )
 
-            emit(Resource.Success(data = localAgenda))
+            val remoteEvents = response.body()?.events?.map { it.toEvent() } ?: emptyList()
+            val remoteReminders = response.body()?.reminders?.map { it.toReminder() } ?: emptyList()
+            val remoteTasks = response.body()?.tasks?.map { it.toTask() } ?: emptyList()
 
-            val isDbEmpty =
-                localEvents.isEmpty() && localTasks.isEmpty() && localReminders.isEmpty()
-            val shouldJustLoadFromCache = !isDbEmpty && !fetchFromRemote
+            remoteEvents + remoteTasks + remoteReminders
 
-            if (shouldJustLoadFromCache) {
-                emit(Resource.Loading(false))
-                return@flow
-            }
+        } catch (e: HttpException) {
+            return AgendaResult.Error(e.message ?: "Invalid Response")
+        } catch (e: IOException) {
+            return AgendaResult.Error(e.message ?: "Couldn't reach server")
+        }
+
+        remoteAgenda.let { agenda ->
+            db.eventDao.deleteEvents()
+            db.eventDao.deleteAttendees()
+            db.reminderDao.deleteReminders()
+            db.taskDao.deleteTasks()
```

Comment:
You only fetch the agenda for a given day, but completely clear all tables here, so the agenda for all other days will be deleted as well